local/local_test.go

```diff
@@ -84,91 +84,236 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
+						h.AssertEq(t, imgOS, daemonOS)
+
+						inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), img.Name())
+						h.AssertNil(t, err)
+						h.AssertEq(t, inspect.Os, daemonOS)
+						h.AssertEq(t, inspect.Architecture, "amd64")
+						h.AssertEq(t, inspect.RootFS.Type, "layers")
+
+						h.AssertEq(t, img.Found(), true)
+					})
 				})
-			})
 
-			when("base image and daemon os/architecture match", func() {
-				it("uses the base image architecture/OS", func() {
-					img, err := local.NewImage(newTestImageName(), dockerClient, local.FromBaseImage(runnableBaseImageName))
-					h.AssertNil(t, err)
-					h.AssertNil(t, img.Save())
-					defer h.DockerRmi(dockerClient, img.Name())
+				when("base image and daemon architecture do not match", func() {
+					it("uses the base image architecture/OS", func() {
+						armBaseImageName := "busybox@sha256:50edf1d080946c6a76989d1c3b0e753b62f7d9b5f5e66e88bef23ebbd1e9709c"
+						expectedArmArch := "arm64"
+						expectedOSVersion := ""
+						if daemonOS == "windows" {
+							// windows/arm nanoserver image
+							armBaseImageName = "mcr.microsoft.com/windows/nanoserver@sha256:29e2270953589a12de7a77a7e77d39e3b3e9cdfd243c922b3b8a63e2d8a71026"
+							expectedArmArch = "arm"
+							expectedOSVersion = "10.0.17763.1040"
+						}
 
-					imgOS, err := img.OS()
-					h.AssertNil(t, err)
-					h.AssertEq(t, imgOS, daemonOS)
+						h.PullIfMissing(t, dockerClient, armBaseImageName)
 
-					inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), img.Name())
-					h.AssertNil(t, err)
-					h.AssertEq(t, inspect.Os, daemonOS)
-					h.AssertEq(t, inspect.Architecture, "amd64")
-					h.AssertEq(t, inspect.RootFS.Type, "layers")
+						img, err := local.NewImage(
+							newTestImageName(),
+							dockerClient,
+							local.FromBaseImage(armBaseImageName),
+						)
+						h.AssertNil(t, err)
+						h.AssertNil(t, img.Save())
+						defer h.DockerRmi(dockerClient, img.Name())
 
-					h.AssertEq(t, img.Found(), true)
+						imgArch, err := img.Architecture()
+						h.AssertNil(t, err)
+						h.AssertEq(t, imgArch, expectedArmArch)
+
+						imgOSVersion, err := img.OSVersion()
+						h.AssertNil(t, err)
+						h.AssertEq(t, imgOSVersion, expectedOSVersion)
+					})
 				})
 			})
 
-			when("base image and daemon architecture do not match", func() {
-				it("uses the base image architecture", func() {
-					armBaseImageName := "arm64v8/busybox@sha256:50edf1d080946c6a76989d1c3b0e753b62f7d9b5f5e66e88bef23ebbd1e9709c"
-					expectedArmArch := "arm64"
-					expectedOSVersion := ""
-					if daemonOS == "windows" {
-						// this nanoserver windows/arm image exists and pulls. Not sure whether it works for anything else.
-						armBaseImageName = "mcr.microsoft.com/windows/nanoserver@sha256:29e2270953589a12de7a77a7e77d39e3b3e9cdfd243c922b3b8a63e2d8a71026"
-						expectedArmArch = "arm"
-						expectedOSVersion = "10.0.17763.1040"
-					}
+			when("#WithPlatform", func() {
```

Comment:
Do we need a set of tests like this the test the combination of `WithPlatform` and `WithPreviousImage`?

Reply:
New test cases: 
local: https://github.com/buildpacks/imgutil/pull/91/files#diff-088ed6061849ec1bfbe08e88f1b8da21e3569421ce717bb8b01e4b94e5523c63R322
remote: https://github.com/buildpacks/imgutil/pull/91/files#diff-55e016054597444d1031ab5c620b77bdf8118f0ee1f82888a2b851850e151726R439